src/tweedle/fields/fp.rs

```diff
@@ -569,6 +553,150 @@ impl Field for Fp {
         CtOption::new(tmp, !self.ct_eq(&Self::zero()))
     }
 
+    fn pow_vartime<S: AsRef<[u64]>>(&self, exp: S) -> Self {
```

Comment:
This is provided by `Field` already, right?

Reply:
Yes, but the `ff` implementation doesn't skip squarings before finding a one, and the existing impl here did, so I assumed you wanted this behaviour.